src/utils/transition-manager.js

```diff
@@ -65,9 +73,17 @@ export default class TransitionManager {
     if (this._isTransitionEnabled(nextProps)) {
       const startProps = Object.assign({}, currentProps,
         this.state.interruption === TRANSITION_EVENTS.SNAP_TO_END ?
-        this.state.endProps : this.state.propsInTransition
+          this.state.endProps : this.state.propsInTransition
       );
 
+      const currentTime = Date.now();
+      if (this.state.interruption === TRANSITION_EVENTS.UPDATE) {
+        const x0 = (currentTime - this.state.startTime) / this.state.duration;
+        nextProps.transitionDuration = this.state.duration - (currentTime - this.state.startTime);
```

Comment:
Do not mutate an argument object - this may cause unexpected side effects for whoever calls this method.